Content.Server/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,178 @@
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly PricingSystem _pricing = default!;
+        [Dependency] private readonly ShuttleSystem _shuttle = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly MapLoaderSystem _map = default!;
+        [Dependency] private readonly ShipyardConsoleSystem _shipyardConsole = default!;
+
+        public MapId? ShipyardMap { get; private set; }
+        private float _shuttleIndex;
+        private const float ShuttleSpawnBuffer = 1f;
+        private ISawmill _sawmill = default!;
+        private bool _enabled;
+
+        public override void Initialize()
+        {
+            _enabled = _configManager.GetCVar(CCVars.Shipyard);
+            _configManager.OnValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            _sawmill = Logger.GetSawmill("shipyard");
+            _shipyardConsole.InitializeConsole();
+            SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, ShipyardConsoleComponent component, ComponentInit args)
+        {
+            if (!_enabled)
+                return;
+
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            _configManager.UnsubValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            CleanupShipyard();
+        }
+
+        private void SetShipyardEnabled(bool value)
+        {
+            if (_enabled == value)
+                return;
+
+            _enabled = value;
+
+            if (value)
+            {
+                SetupShipyard();
+            }
+            else
+            {
+                CleanupShipyard();
+            }
+        }
+
+        /// <summary>
+        /// Adds a ship to the shipyard, calculates its price, and attempts to ftl-dock it to the given station
+        /// </summary>
+        /// <param name="stationUid">The ID of the station to dock the shuttle to</param>
+        /// <param name="shuttlePath">The path to the shuttle file to load. Must be a grid file!</param>
+        public void PurchaseShuttle(EntityUid? stationUid, string shuttlePath, out ShuttleComponent? vessel)
+        {
+            if (!TryComp<StationDataComponent>(stationUid, out var stationData) || !TryComp<ShuttleComponent>(AddShuttle(shuttlePath), out var shuttle))
+            {
+                vessel = null;
+                return;
+            }
+
+            var targetGrid = _station.GetLargestGrid(stationData);
+
+            if (targetGrid == null)
+            {
+                vessel = null;
+                return;
+            }
+
+            var price = _pricing.AppraiseGrid(shuttle.Owner, null);
```

Comment:
Why is the price calculated from scratch instead of using the prototype when you have a prototype for it.

Reply:
this is for calculating the real price against the stated prototype price. for now it is just output in to the console for admin review, and used in testing when creating and testing that a new shuttle will work through the system, and potentially used later to verify that a grid is not being purchased for less than it's worth since sloth suggested a limit on the number of purchasable shuttles per round and having the ability to sell the shuttle to free up a deed slot to purchase a new one.

Reply:
Implement a test for this instead of using appraisal. Admins and devs can pretty easily just use the actual appraise grid command when verifying.